test/ex_doc/autolink_test.exs

```diff
@@ -76,130 +81,131 @@ defmodule ExDoc.AutolinkTest do
         {{:function, Foo, :foo, 1}, true}
       ])
 
-      assert autolinked(~t"foo/1", current_module: Foo) == "#foo/1"
-      assert_unchanged(~t"bar/1", current_module: Foo)
+      assert autolink("foo/1", current_module: Foo) == ~m"[`foo/1`](#foo/1)"
+      assert_unchanged("bar/1", current_module: Foo)
     end
 
     test "elixir callback" do
-      assert autolinked(~t"c:GenServer.handle_call/3") ==
-               "https://hexdocs.pm/elixir/GenServer.html#c:handle_call/3"
+      assert autolink("c:GenServer.handle_call/3") ==
+               ~m"[`c:GenServer.handle_call/3`](https://hexdocs.pm/elixir/GenServer.html#c:handle_call/3)"
+
+      # TODO: there should be no `c:` in the link _text_!
+      # assert autolink("c:GenServer.handle_call/3") ==
+      #          ~m"[`GenServer.handle_call/3`](https://hexdocs.pm/elixir/GenServer.html#c:handle_call/3)"
```

Comment:
I'll fix it in the next PR, I thought I'd improve the tests to expose this regression first.